### PR TITLE
Update mypy to 1.4.1

### PR DESCRIPTION
**Pull request by malfet**
Mostly fixes for PEP-484 violation (i.e. when default arg is set to None, but type is not annotated as optional)
Plus few real fixes:
  - Add missing `_get_upgraders_entry_map` to `torch/_C/__init__.pyi`
  - Add missing return statement to `torch._export. deserialize_graph`
  - Fix error message in `torch.ao.ns.fx.weight_utils.get_lstm_mod_weights`
  - 
TODO (in followup PR):
  - Fix erroneous `isinstance` check in `torch/ao/quantization/_pt2e/qat_utils.py`

### CODE REVIEW

**Review by kit1980 — APPROVED** [review]

File: .lintrunner.toml [hunk 1]
```diff
@@ -96,6 +96,7 @@ include_patterns = [
     'test/test_type_hints.py',
     'test/test_type_info.py',
     'test/test_utils.py',
+    'github/.scripts/*.py',
```
Comment (ZainRizvi) [inline .lintrunner.toml:99]: Maybe future proof it via:

```suggestion
    'github/.scripts/**/*.py',
```

**Review by ZainRizvi — APPROVED** [review]

**Review by huydhn — APPROVED** [review]
Having newer mypy version would help address https://github.com/pytorch/pytorch/issues/98260

**Review by thiagocrepaldi — APPROVED** [review]
nice!

**Review by aaronenyeshi — APPROVED** [review]
Profiler parts LGTM!

**Comment by malfet** [comment]
> @pytorchbot -c ghfirst -m "need to be reverted in order to sync with internal changes"

@atalman bot is not that smart

**Pull request closed without merging**